web-app/src/components/DipMap.js

```diff
@@ -745,6 +745,110 @@ export default class DipMap extends React.Component {
 					this.debugCount("updateMap/hidProvince");
 				}
 			}
+
+
+			//Here we check each non-SC and non-Sea territory. If all surrounding SCs are of the same power and none is "Neutral", colour them that power.
+			//Get all nodes, disqualify Sea and SC, and per node collect the edges in an array.
+			for (let prov in this.state.variant.Properties.Graph.Nodes) {
+				const node = this.state.variant.Properties.Graph.Nodes[prov];
+				if (!node.SC && node.Subs[""].Flags.Land) {
+					const borderprovs = [];
+					for (let edge in node.Subs[""].Edges) {
+						const edgenode = this.state.variant.Properties.Graph.Nodes[edge];
+						if (typeof edgenode !== "undefined") {
+							if (edgenode.SC) {
+								borderprovs.push(SCs[edgenode.Name]);
+							}
+						}
+					}
```

Comment:
Use `map` here:
```
const borderProvs = node.Subs[""].Edges.map((edge) => nodes.[edge]).map((edgeNode) => edgeNode && edgeNode.SC);
```

Reply:
This code may not be correct. I'm just eye-balling it. But it should be pretty close. Also make sure to use camel-case, `edgeNode` rather than `edgenode`

Reply:
While this might work, it gets all kind of errors - and the whole reason to have a look at this from your perspective was to make it better :). I can't "learn" what I don't know... 

There's an error for the map function because of the [""] implementation, it says Edges.. has no function.